Lib/asyncore.py

```diff
@@ -127,7 +127,8 @@ def poll(timeout=0.0, map=None):
         map = socket_map
     if map:
         r = []; w = []; e = []
-        for fd, obj in list(map.items()):
+        map_copy = dict(map)
```

Comment:
I wonder if it is better to copy only the readable and writable objects, instead of the whole map.

We can do:

    pending = {}
    for fd, obj in map.items():
       ...
       if is_r or is_w:
          pending[fd] = obj

And later handle events only on objects from the pending dict. This may be faster if only some of the objects are active.